__tests__/utils.test.js

```diff
@@ -1,120 +1,126 @@
+        const filledObject = fillMissingKeys(myObject, defaultObject)
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second).toBe(2)
+        expect(filledObject.third).toBe(5)
+        expect(filledObject.fourth).toBe(4)
+    })
+
+    test('check that fillMissingKeys works with empty objects', () => {
+        const myObject = {}
+        const defaultObject = {first: 1, second: 2}
+        const filledObject = fillMissingKeys(myObject, defaultObject)
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second).toBe(2)
+
+        const myObject2 = {first: 1, second: 2}
+        const defaultObject2 = {}
+        const filledObject2 = fillMissingKeys(myObject2, defaultObject2)
+
+        expect(filledObject2.first).toBe(1)
+        expect(filledObject2.second).toBe(2)
+
+    })
+
+    test('check that fillMissingKeys works on a copy', () => {
+        const myObject = {}
+        const defaultObject = {first: 1, second: 2}
+        const filledObject = fillMissingKeys(myObject, defaultObject)
+
+        expect(myObject.first).toBe(undefined)
+        expect(myObject.second).toBe(undefined)
+    })
+
+    test('check that fillMissingKeys supports nested objects', () => {
+        let myObject = {}
+        let defaultObject = {first: 1, second: {second_1: 2, second_2: 3}}
+        let filledObject = fillMissingKeys(myObject, defaultObject)
+        const deep_copy = true
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second.second_1).toBe(2)
+        expect(filledObject.second.second_2).toBe(3)
+
+        myObject = {second: {second_1: 4}}
+        defaultObject = {first: 1, second: {second_1: 2, second_2: 3}}
+        filledObject = fillMissingKeys(myObject, defaultObject, deep_copy)
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second.second_1).toBe(4)
+        expect(filledObject.second.second_2).toBe(3)
+
+        myObject = {
+            second: {
+                second_1: 4,
+                second_2: {
+                    second_2_1: 5
+                }
+            }
         }
-      }
-    }
-    defaultObject = {first: 1, second: {second_1: 2, second_2: 3}}
-    filledObject = fillMissingKeys(myObject, defaultObject, deep_copy)
-
-    expect(filledObject.first).toBe(1)
-    expect(filledObject.second.second_1).toBe(4)
-    expect(filledObject.second.second_2.second_2_1).toBe(5)
-  })
+        defaultObject = {first: 1, second: {second_1: 2, second_2: 3}}
+        filledObject = fillMissingKeys(myObject, defaultObject, deep_copy)
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second.second_1).toBe(4)
+        expect(filledObject.second.second_2.second_2_1).toBe(5)
+    })
+
+    test('check that mid-color is calculated correctly', () => {
+        let midColor = calculateMidColor('#ffffff', '#22a6b3');
+
+        expect(midColor).toBe('#90D2D9');
```

Comment:
Unit test for calculateMidColor, looks good

Reply:
thanks 👍